src/Data/Nat/Properties.agda

```diff
@@ -2101,31 +2101,43 @@ n≤′m+n (suc m) n = ≤′-step (n≤′m+n m n)
 -- Properties of _≤″_ and _<″_
 ------------------------------------------------------------------------
 
-m<ᵇn⇒1+m+[n-1+m]≡n : ∀ m n → T (m <ᵇ n) → suc m + (n ∸ suc m) ≡ n
-m<ᵇn⇒1+m+[n-1+m]≡n m n lt = m+[n∸m]≡n (<ᵇ⇒< m n lt)
+-- equivalence of  _≤″_ to _≤_
 
-m<ᵇ1+m+n : ∀ m {n} → T (m <ᵇ suc (m + n))
-m<ᵇ1+m+n m = <⇒<ᵇ (m≤m+n (suc m) _)
+≤⇒≤″ : _≤_ ⇒ _≤″_
+≤⇒≤″ = less-than-or-equal ∘ m+[n∸m]≡n
 
-<ᵇ⇒<″ : T (m <ᵇ n) → m <″ n
-<ᵇ⇒<″ {m} {n} leq = less-than-or-equal (m+[n∸m]≡n (<ᵇ⇒< m n leq))
+<⇒<″ : _<_ ⇒ _<″_
+<⇒<″ = ≤⇒≤″
 
-<″⇒<ᵇ : ∀ {m n} → m <″ n → T (m <ᵇ n)
-<″⇒<ᵇ {m} (<″-offset k) = <⇒<ᵇ (m≤m+n (suc m) k)
+≤″⇒≤ : _≤″_ ⇒ _≤_
+≤″⇒≤ (≤″-offset k) = m≤m+n _ k
 
 -- equivalence to the old definition of _≤″_
 
-≤″-proof : ∀ {m n} (le : m ≤″ n) → let less-than-or-equal {k} _ = le in m + k ≡ n
+≤″-proof : (le : m ≤″ n) → let less-than-or-equal {k} _ = le in m + k ≡ n
 ≤″-proof (less-than-or-equal prf) = prf
 
--- equivalence to _≤_
+-- yielding corresponding proof and `pattern` aliases for _≤″_/_<_
 
-≤″⇒≤ : _≤″_ ⇒ _≤_
-≤″⇒≤ {zero}  (≤″-offset k) = z≤n {k}
-≤″⇒≤ {suc m} (≤″-offset k) = s≤s (≤″⇒≤ (≤″-offset k))
+≤-proof : m ≤ n → ∃ λ k → m + k ≡ n
+≤-proof le = _ , ≤″-proof (≤⇒≤″ le)
 
-≤⇒≤″ : _≤_ ⇒ _≤″_
-≤⇒≤″ = less-than-or-equal ∘ m+[n∸m]≡n
+pattern ≤-offset k = k , refl
+pattern <-offset k = ≤-offset k
```

Comment:
We don't traditionally introduce pattern synonyms in the `Properties` files. If we want to keep them, they should live in `Data.Nat.Base`... 

However, I might vote for removing them. The presence of `refl` in a proof is usually quite indicative of what's going on so we'd be losing that. Also without any type annotations on the pattern synonyms, even if you look at them, it's not quite clear what they are constructing....

Reply:
OK, two or three things:
* `style-guide` addition: pattern synonyms should be in `Base` not `Properties
* naming, as with your comment below on `≤-proof` below; some of the changes here are in some sense knock-on changes for the work previously done on trying to constrain/eliminate use of `_≤″_`, including that relation's pattern synonyms
* the use of `refl` in specialising a constructor was a sufficiently common idiom that it seemed worthwhile to reify it as a synonym. But you're right in that perhaps that's not necessary... in which case perhaps also `divides-refl` of `Data.Nat.Divisibility.Core` (introduced in #2013 ) should be removed?

Reply:
> However, I might vote for removing them. The presence of `refl` in a proof is usually quite indicative of what's going on so we'd be losing that. Also without any type annotations on the pattern synonyms, even if you look at them, it's not quite clear what they are constructing....

Most recent commit removes the new synonyms, but not (yet!)  the `Data.Nat.Base.≤″-offset` one?

Reply:
> But you're right in that perhaps that's not necessary... in which case perhaps also divides-refl of Data.Nat.Divisibility.Core (introduced in https://github.com/agda/agda-stdlib/pull/2013 ) should be removed?

I think `divides-refl` is slightly less problematic, because at least it has `refl` in the name...

> Most recent commit removes the new synonyms, but not (yet!) the Data.Nat.Base.≤″-offset one?

Hmm yes, they should probably be deprecated as well, (as one of the main points of this PR was to remove them right?)

Reply:
The *main* point is to (almost!) reach the end of a line where any *use* of `_≤″_` is removed entirely... while leaving the definition, and its equivalence with `_≤_` in `Data.Nat.*` as a legacy fossil relic.

But I couldn't quite get rid of it entirely: `Data.Fin.Base.from≤″` uses it (but I hope that a merge of #2257 and then the `breaking` change to `Data.Nat.DivMod{.WithK}` discussed there would deliver the final final blow...)

Reply:
Actually,  I think I was too hasty in responding to your call to remove the new pattern synonyms, precisely *because* they encapsulate the ability to compute the offset `k` from a proof of `m ≤ n` *without having to invoke* either `_≤″_` *or* its own pattern synonyms `less-than-or-equal` or `≤″-offset` outside `Data.Nat.Base` or `Data.Nat.Properties`... with the latter only being necessary on the model of the debate about whether these (proxy) inversions belong in `base` or `Properties`. 

More thought required!

Reply:
Smilarly to @Taneb 's comment below, now almost all uses of the lemma removed in favour of direct pattern matching, a 'guarded' definition of monus, and indeed finally along with deprecation of the pattern synonyms... and inlining the proof of `s<″s⁻¹` in the definition of `Data.Fin.Base.from≤″`.

Reply:
Inlining wasn't possible, because of the need to prove lemmas about `from≤″` in `Data.Fin.Properties` but we're nearly all the way there!